### PR TITLE
Add report doc loader

### PR DESCRIPTION
**Pull request by HarryMytilinaios**
Added custom workaround to show loader in creating pdf reports of exposure analysis (currently waiting about 20 seconds for the report to be generated)

How to test the feature:
Select exposure analysis and click the create report button on the exposure analysis actions below in the sidebar.

### CODE REVIEW

File: frontend/src/components/Common/ReportDialog/index.tsx [hunk 1]
```diff
@@ -150,7 +192,11 @@ const ReportDialog = memo(
           </span>
           <Button className={classes.actionButton} variant="outlined">
             <PDFDownloadLink document={reportDoc} fileName={getPDFName}>
-              {({ loading }) => (loading ? 'Loading document...' : 'Download')}
+              {({ loading }) =>
+                loading || documentIsLoading
+                  ? 'Loading document...'
```
Comment (ericboucher) [inline frontend/src/components/Common/ReportDialog/index.tsx:197]: These texts should get translated if possible

File: frontend/src/components/Common/ReportDialog/index.tsx [hunk 2]
```diff
@@ -107,6 +119,29 @@ const ReportDialog = memo(
...
+      }
+      return (
+        <Box className={classes.documentLoadingContainer}>
+          <CircularProgress size={100} />
+          <Typography
+            className={classes.documentLoadingText}
+            variant="body1"
+            component="span"
+          >
+            {t('Loading document...')}
```
Comment (ericboucher) [inline frontend/src/components/Common/ReportDialog/index.tsx:134]: Let's remove the `...` from the trabslation function

File: frontend/src/components/MapView/LeftPanel/AnalysisPanel/index.tsx [hunk 3]
```diff
@@ -670,9 +670,21 @@ function AnalysisPanel({
     return (
       <Box className={classes.exposureAnalysisLoadingContainer}>
         <CircularProgress size={100} />
+        <Typography
+          className={classes.exposureAnalysisLoadingText}
+          variant="body1"
+          component="span"
+        >
+          {t('Loading Exposure Analysis Data...')}
```
Comment (ericboucher) [inline frontend/src/components/MapView/LeftPanel/AnalysisPanel/index.tsx:678]: let's move the `...` after the translation

**Comment by wadhwamatic** [comment]
Thanks @HarryMytilinaios this is very needed. The loader itself was 'choppy' for me for the first several seconds. Are you seeing this effect?

https://user-images.githubusercontent.com/3343536/228328676-ace48499-05e4-48e6-9511-ef7b39df9f80.mov

**Comment by ericboucher** [comment]
Yes @wadhwamatic we're seeing the choppiness as well. That's because the report generation is taking all the compute power. We're looking into having a simple blinking 3 dots instead to make it easier on the CPU.

**Comment by HarryMytilinaios** [comment]
Added only the `Loading Document...` as a loading text. The problem however is still there. The dots loading and the overall app is laggy when the document is loading for some seconds. This is caused by `@react-pdf/renderer` component library. The problem here is that the document is really large and it is created in the frontend and especially in javascript's main thread. I am not sure what is the correct approach on this. One possible workaround would be to render only some pages (e.g. The first 5 for example), and then prompt the user to download the whole file, I think that would prevent the lagginess at some extent.

File: frontend/src/components/Common/ReportDialog/index.tsx (hunk 2 above)
Comment (HarryMytilinaios) [reply frontend/src/components/Common/ReportDialog/index.tsx]: Updated

File: frontend/src/components/Common/ReportDialog/index.tsx (hunk 1 above)
Comment (HarryMytilinaios) [reply frontend/src/components/Common/ReportDialog/index.tsx]: Updated

**Comment by wadhwamatic** [comment]
@HarryMytilinaios and @ericboucher. The text approach definitely works better. I think this is an extreme case with the number of rows of data we have being rendered. In the future, we could also limit / filter out unimportant data. But this works for now - thanks a lot.

A server-side approach sounds like the way to go given we know we'll have users who won't have high-end devices.

File: frontend/src/components/Common/ReportDialog/index.tsx [hunk 4]
```diff
@@ -107,6 +118,32 @@ const ReportDialog = memo(
...
+          <Typography
+            className={classes.documentLoaderText}
+            variant="body1"
+            component="span"
+          >
+            {t('Loading document')}
+          </Typography>
+          <span className={classes.documentLoaderDot}>.</span>
+          <span className={classes.documentLoaderDot}>.</span>
+          <span className={classes.documentLoaderDot}>.</span>
```
Comment (ericboucher) [inline frontend/src/components/Common/ReportDialog/index.tsx:136]: Let's wrap this in a small reusable component with a self-explanatory name please?

File: frontend/src/components/Common/ReportDialog/reportDoc.tsx [hunk 5]
```diff
@@ -89,7 +89,7 @@ const ReportDoc = memo(
     columns,
   }: ReportDocProps) => {
     if (mapImage === null) {
-      return <></>;
+      return <Document />;
```
Comment (ericboucher) [inline frontend/src/components/Common/ReportDialog/reportDoc.tsx:92]: I think this was causing an error before. See https://github.com/WFP-VAM/prism-app/pull/624
Ask @echaidemenos for more info if needed

**Review by ericboucher — APPROVED** [review]

File: frontend/src/components/Common/ReportDialog/index.tsx (hunk 4 above)
Comment (ericboucher) [reply frontend/src/components/Common/ReportDialog/index.tsx]: I meant move the blinkingDots into their own component ;) @HarryMytilinaios

File: frontend/src/components/Common/ReportDialog/reportDoc.tsx (hunk 5 above)
Comment (HarryMytilinaios) [reply frontend/src/components/Common/ReportDialog/reportDoc.tsx]: Yes you are right @ericboucher. This was causing an error but adding the `<></>` component was causing other console errors. I've made a fix that removes the if statement at the root of the report doc component and prevents the auto downloading of the pdf when mounted and not opened. I'll make some tests in different browsers and windows firefox that we had this issue before to see that everything is going as expected

File: frontend/src/components/Common/ReportDialog/index.tsx (hunk 4 above)
Comment (HarryMytilinaios) [reply frontend/src/components/Common/ReportDialog/index.tsx]: Updated

Comment (ericboucher) [reply frontend/src/components/Common/ReportDialog/index.tsx]: Am I missing a commit maybe?

Comment (HarryMytilinaios) [reply frontend/src/components/Common/ReportDialog/index.tsx]: Ohhh I understood to move the whole component... Sorry for the misunderstanding chaning it

**Pull request merged**